### PR TITLE
discord.js v13 -> v14.13.0

### PR DESCRIPTION
**Pull request by juddisjudd**
- General required refactoring of code from v13 methods to v14.
- Further refactoring done for consistency and to minimize code in certain areas.
- Various packages updated/removed as well to align with discord.js update.
- Every command and event has been tested in a testing environment discord and all seems to work as intended.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -10,21 +10,18 @@
...
   "devDependencies": {
-    "@types/node": "*",
-    "@types/node-fetch": "^2.6.1",
-    "concurrently": "^7.1.0",
-    "eslint": "^8.13.0",
-    "nodemon": "^2.0.15",
-    "prettier": "2.6.2",
-    "ts-node": "^10.7.0",
-    "typescript": "^4.6.3"
+    "@types/node": "20.5.4",
```
Comment (LukeWasTakenn) [inline package.json]: Should be using LTS nodejs (18.17.1)

File: src/commands/docs.ts [hunk 2]
```diff
@@ -14,21 +13,24 @@ const Docs: Command = {
...
+    const resource = resourceOption?.value as string;
+    await sendDocumentationEmbed(interaction, resource);
   },
 };
 
+const sendDocumentationEmbed = async (interaction: CommandInteraction, resource: string) => {
+  const docsEmbed = new EmbedBuilder()
+    .setTitle('Documentation')
+    .setDescription('Please read the documentation thoroughly and carefully.')
+    .setColor(0x26e0a5)
```
Comment (LukeWasTakenn) [inline src/commands/docs.ts]: Does color not support strings? Would rather that than this weird hex value that you can't tell what it is off of the first glace.

File: src/commands/ox.ts [hunk 3]
```diff
@@ -1,40 +1,45 @@
...
-    },
-  ],
-};
-
 const Ox: Command = {
   data: new SlashCommandBuilder().setName('ox').setDescription('About Overextended'),
-  run: async (interaction) => await interaction.reply({ embeds: [OxEmbed] }),
+  execute: async (interaction) => {
+    await interaction.reply({ embeds: [oxEmbed()] });
+  },
```
Comment (LukeWasTakenn) [inline src/commands/ox.ts]: What's up with some commands using `run` and others using `execute`? Also the inconsistencies between using normal and arrow functions.

File: src/index.ts [hunk 4]
```diff
@@ -1,19 +1,34 @@
...
-Bot.on('guildMemberRemove', async (member) => await onMemberRemove(member));
-Bot.on('guildBanAdd', async (ban) => await onMemberBan(ban));
+Bot.on(Events.GuildAuditLogEntryCreate, async (auditLogEntry, guild) => {
+  if (auditLogEntry.action === AuditLogEvent.MemberBanAdd) {
+    await onMemberBan(auditLogEntry, guild);
+  } else if (auditLogEntry.action === AuditLogEvent.MemberBanRemove) {
+    await onMemberUnban(auditLogEntry, guild);
+  } else if (auditLogEntry.action === AuditLogEvent.MemberKick) {
+    await onMemberRemove(auditLogEntry, guild);
+  }
```
Comment (LukeWasTakenn) [inline src/index.ts]: switch would be a lot nicer here

**Review by LukeWasTakenn — CHANGES_REQUESTED** [review]

File: src/commands/docs.ts (hunk 2 above)
Comment (juddisjudd) [reply src/commands/docs.ts]: Misread documentation, changed to regular web friendly hex code.

File: src/index.ts (hunk 4 above)
Comment (juddisjudd) [reply src/index.ts]: been doing too much python lately, will change to switch

File: src/commands/ox.ts (hunk 3 above)
Comment (juddisjudd) [reply src/commands/ox.ts]: Will make consistent, mostly was due to multiple re-writes of this shit.

**Review by LukeWasTakenn — APPROVED** [review]

**Pull request merged**